scala-package/core/src/main/scala/org/apache/mxnet/NDArray.scala

```diff
@@ -405,15 +405,13 @@ object NDArray extends NDArrayBase {
    * @param stop End of interval.
    * @param step Spacing between values. The default step size is 1.
    * @param repeat Number of times to repeat each element. The default repeat count is 1.
-   * @param infer_range
-   *          When set to True, infer the stop position from the start, step,
-   *          repeat, and output tensor size.
    * @param ctx Device context. Default context is the current default context.
    * @param dType The data type of the `NDArray`. The default datatype is `DType.Float32`.
    * @return NDArray of evenly spaced values in the specified range.
    */
-  def arange(start: Float, stop: Option[Float], step: Float,
-             repeat: Int, ctx: Context, dType: DType): NDArray = {
+  def arange(start: Float, stop: Option[Float] = None, step: Float = 1.0f,
```

Comment:
Unless you are certain that this param is not going to be used, please place it at the end of the param list and rename it as inferRange follows Java camel case.

Reply:
It's not being used now. Do you mean we still include it now in case it'll be used later?

Reply:
According to https://github.com/apache/incubator-mxnet/pull/12064#discussion_r211408234 `infer_range` is only applicable to symbolic API and was meant to be removed from ndarray API.

Reply:
Add @taliesinb here to bring more context since I am not very familiar with it